joss/paper.md

```diff
@@ -0,0 +1,151 @@
+would be useful to researchers and students alike, such as sophisticated construction and manipulation
+algorithms. Moreover, Python is a popular tool for students and researchers, meaning the availability
+of a high-quality software package is likely to encourage the further exploration of these structures
+in the academic community.
+
+# The `automata` package
+
+`automata` is a Python package for the manipulation and simulation of automata from the families listed above.
+The API is designed to mimic the formal mathematical description of each automata using built-in Python data structures. As a popular high-level language, Python enables greater flexibility and easy-of-use that is difficult
+to achieve with a low-level language (e.g., Rust). Algorithms in the package have been optimized for
+performance against benchmarks from tasks arising in research. In addition, Python allows for
+greater optimization by the integration of lower-level technologies (e.g., Cython), while still
+retaining the same high-level API, allowing for integration of more performant features as-needed by
+the user base. The package also has native display integration with Jupyter notebooks, enabling
+easy visualization.
+
+Of note are some sophisticated and useful algorithms implemented in the package for finite-state automata:
+
+- An optimized version of the Hopcroft-Karp algorithm to determine whether two deterministic finite automata (DFA) are equivalent [@AlmeidaMR10].
+
+- Thompson's algorithm for converting regular expressions to equivalent nondeterministic finite automata (NFA) [@AhoSU86].
+
+- Hopcroft's algorithm for DFA minimization [@Yingjie09].
+
+- A specialized algorithm for directly constructing a state-minimal DFA accepting a given
+finite language [@mihov_schulz_2019].
+
+- A specialized algorithm for directly constructing a minimal DFA recognizing strings containing
+a given substring [@Knuth77].
+
+To the author's knowledge, this is the only Python package implementing a number of the algorithms stated above.
+
+`automata` was designed around existing theoretical models of automata, for use by both
+mathematically-oriented researchers and in educational contexts. The
+included functionality for parsing regular expressions and manipulating finite-state
+machines enables fast and accessible exploration of these structures by researchers.
+On the educational side, the package includes visualization logic that allows students to
+interact with these structures in an exploratory manner, and has already seen usage in
+undergraduate courses. `automata` has already been cited in publications [@Erickson23], with more
+to come as the package becomes more popular.
+
+`automata` has seen a large number of contributions by external contributors and wide adoption,
+demonstrating the demand for a high-quality Python package providing these features. The code is
+well maintained, including a comprehensive test suite and type annotations, meaning new features
```

Comment:
@eliotwrobson If I'm not mistaken, I think "well maintained" needs to be hyphenated (i.e. "well-maintained").

(the academic context of this paper is my only excuse to nitpick the grammar 😅)

Reply:
Good catch! Done here: https://github.com/caleb531/automata/pull/163/commits/ef46c654e4d32e72519fe31bbf85a46121a0f737